tests/conftest.py

```diff
@@ -111,49 +111,46 @@ def mock_session() -> CachedSession:
     This uses a temporary SQLite db stored in ``/tmp``, which will be removed after the fixture has
     exited.
     """
-    with NamedTemporaryFile(suffix='.db') as temp:
-        session = CachedSession(
-            db_path=temp.name,
-            backend='sqlite',
-            allowable_methods=ALL_METHODS,
-            suppress_warnings=True,
-        )
-        yield mount_mock_adapter(session)
+    session = CachedSession(
+        db_path=tmpdir / "test.db",
+        backend='sqlite',
+        allowable_methods=ALL_METHODS,
+        suppress_warnings=True,
+    )
+    yield mount_mock_adapter(session)
 
 
 @pytest.fixture(scope='function')
-def tempfile_session() -> CachedSession:
+def tempfile_session(tmpdir) -> CachedSession:
     """Get a CachedSession using a temporary SQLite db"""
-    with NamedTemporaryFile(suffix='.db') as temp:
-        session = CachedSession(
-            cache_name=temp.name,
-            backend='sqlite',
-            allowable_methods=ALL_METHODS,
-            suppress_warnings=True,
-        )
-        yield session
+    session = CachedSession(
+        cache_name=tmpdir / "test.db",
+        backend='sqlite',
+        allowable_methods=ALL_METHODS,
+        suppress_warnings=True,
+    )
+    yield session
 
 
 @pytest.fixture(scope='function')
-def tempfile_path() -> CachedSession:
+def tempfile_path(tmpdir) -> CachedSession:
     """Get a tempfile path that will be deleted after the test finishes"""
-    with NamedTemporaryFile(suffix='.db') as temp:
-        yield temp.name
+    yield tmpdir / "test.db"
 
 
 @pytest.fixture(scope='function')
-def installed_session() -> CachedSession:
+def installed_session(tmpdir) -> CachedSession:
```

Comment:
Good call, I didn't know about the `tmpdir` fixture. I'm surprised `NamedTemporaryFile` didn't work on Windows, though. Can you post the full traceback you got?

Reply:
Well, wouldn't you know, I can't reproduce the issue now...

Reply:
No worries, I like the `tmpdir` version better anyway. I'm just interested if there's more platform-specific I/O behavior that I need to be aware of.